src/transport.c

```diff
@@ -130,30 +138,50 @@ debugdump(LIBSSH2_SESSION * session,
 
 static int
 decrypt(LIBSSH2_SESSION * session, unsigned char *source,
-        unsigned char *dest, ssize_t len)
+        unsigned char *dest, ssize_t len, int firstlast)
 {
     struct transportpacket *p = &session->packet;
     int blocksize = session->remote.crypt->blocksize;
 
     /* if we get called with a len that isn't an even number of blocksizes
-       we risk losing those extra bytes */
-    assert((len % blocksize) == 0);
+       we risk losing those extra bytes. AAD is an exception, since those first
+       few bytes aren't encrypted so it throws off the rest of the count. */
+    if(!CRYPT_FLAG_L(session, PKTLEN_AAD))
+        assert((len % blocksize) == 0);
+
+    while(len > 0) {
+        /* normally decrypt up to blocksize bytes at a time */
+        int decryptlen = MIN(blocksize, len);
```

Comment:
```suggestion
        int decryptlen = LIBSSH2_MIN(blocksize, len);
```
Prefer newly added `LIBSSH2_MAX()`/`LIBSSH2_MIN()`.